app/models/discovery_rule.rb

```diff
@@ -11,6 +11,7 @@ class DiscoveryRule < ApplicationRecord
   validates :hostname, :format => { :with => /\A[a-zA-Z<]/, :message => N_("must start with a letter or ERB.") },
     :allow_blank => true
   validates :hostgroup_id, :presence => true
+  validates :hostgroup, :presence => { :message => N_("does not exist, please check hostgroup-id.") }, :if => :hostgroup_id?
```

Comment:
@lzap the intention here is not to validate the hostgroup_id but to have a custom error message. If we only have `validates :hostgroup, :presence => true`, the error message would be something like - `Hostgroup can't be blank` which I feel will not convey the correct error to the end-users.

Reply:
I don't understand how "Hostgroup can't be blank" is not correct error. But if you insist, go for "must be present". Also remove the "if => :hostgroup_id?" part, that prevents from validating new records.